### PR TITLE
Do not harcode namespace

### PR DESCRIPTION
**Pull request by bclermont**
Is there any reason why documentation use `kube-system`?

I rather keep different apps in separated namespaces.

### CODE REVIEW

File: charts/stash/templates/mutating-webhook.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ webhooks:
 - name: deployment.admission.stash.appscode.com
   clientConfig:
     service:
-      namespace: default
+      namespace: {{ .Release.Namespace }}
```
Comment (tamalsaha) [inline charts/stash/templates/mutating-webhook.yaml]: All changes made in this file are incorrect. The webhook server is registered as an extended apiserver and only accessed through the default/kubernetes service.

File: charts/stash/templates/validating-webhook.yaml [hunk 2]
```diff
@@ -15,7 +15,7 @@ webhooks:
 - name: restic.admission.stash.appscode.com
   clientConfig:
     service:
-      namespace: default
+      namespace: {{ .Release.Namespace }}
```
Comment (tamalsaha) [inline charts/stash/templates/validating-webhook.yaml]: All changes made in this file are incorrect. The webhook server is registered as an extended apiserver and only accessed through the default/kubernetes service.

File: charts/stash/templates/mutating-webhook.yaml (hunk 1 above)
Comment (bclermont) [reply charts/stash/templates/mutating-webhook.yaml]: ok, will revert

File: charts/stash/templates/validating-webhook.yaml (hunk 2 above)
Comment (bclermont) [reply charts/stash/templates/validating-webhook.yaml]: ok, will revert

File: charts/stash/templates/deployment.yaml [hunk 3]
```diff
@@ -19,9 +19,9 @@ spec:
     metadata:
       labels:
         {{- include "stash.labels" . | nindent 8 }}
-{{- if or .Values.annotations (and .Values.criticalAddon (eq .Release.Namespace "kube-system")) }}
+{{- if or .Values.annotations .Values.criticalAddon }}
```
Comment (tamalsaha) [inline charts/stash/templates/deployment.yaml]: critical addons were restricted to kube-system namespace until 1.17 .
https://github.com/kubernetes/kubernetes/issues/60596#issuecomment-559537077

So, you need to check for k8s version before relaxing this. Otherwise pods will not start in < 1.17 release.

File: charts/stash/templates/apiregistration.yaml [hunk 4]
```diff
@@ -99,7 +99,7 @@ apiVersion: rbac.authorization.k8s.io/v1
 kind: RoleBinding
 metadata:
   name: {{ template "stash.fullname" . }}-apiserver-extension-server-authentication-reader
-  namespace: kube-system
+  namespace: {{ .Release.Namespace }}
```
Comment (tamalsaha) [inline charts/stash/templates/apiregistration.yaml]: This is a Role binding and can only created in the namespace of the role which is kube-system. Please revert this change.

File: charts/stash/templates/deployment.yaml (hunk 3 above)
Comment (tamalsaha) [reply charts/stash/templates/deployment.yaml]: Example: https://github.com/stashed/installer/blob/master/charts/stash/templates/mutating-webhook.yaml#L2-L3

**Comment by bclermont** [comment]
@tamalsaha PR updated

**Comment by bclermont** [comment]
I added a commit to force deployment to restart when tls.cert/tls.key are changed in the `kind: Secret`

**Comment by tamalsaha** [comment]
Can you please squash the commits and sign-off?

**Comment by bclermont** [comment]
@tamalsaha done

BTW I had been running this for 2 weeks now

**Comment by bclermont** [comment]
Just added a minor change that convert operator resources into a value

**Comment by bclermont** [comment]
>BTW I had been running this for 1 week now

I just realized it wasn't that but an other fork, testing it

**Comment by bclermont** [comment]
> I just realized it wasn't that but an other fork, testing it

tested

**Comment by bclermont** [comment]
something else not ok and that I should do with this PR?

**Comment by bclermont** [comment]
I had been using this in production on few clusters for > 3 weeks

File: charts/stash/templates/deployment.yaml [hunk 5]
```diff
@@ -7,6 +10,7 @@ metadata:
     {{- include "stash.labels" . | nindent 4 }}
 {{- if .Values.annotations }}
   annotations:
+    checksum/apiregistration.yaml: {{ include (print $.Template.BasePath "/apiregistration.yaml") . | sha256sum }}
```
Comment (tamalsaha) [inline charts/stash/templates/deployment.yaml:13]: Nice trick!

https://helm.sh/docs/howto/charts_tips_and_tricks/#automatically-roll-deployments

**Comment by tamalsaha** [comment]
Thanks a lot for your work on this pr, @bclermont ! I really appreciate this. And sorry for the delay to merge the pr.
